server/filestore.go

```diff
@@ -8040,6 +8049,7 @@ func (fs *fileStore) compact(seq uint64, _ /* noMarkers */ bool) (uint64, error)
 			// Update fss
 			smb.removeSeqPerSubject(sm.subj, mseq)
 			fs.removePerSubject(sm.subj, !noMarkers && fs.cfg.SubjectDeleteMarkerTTL > 0)
+			tombs = append(tombs, msgId{sm.seq, sm.ts})
```

Comment:
Should we not do this if we know we are going to nil out below?
We can check dmap or something to know apriori if we will need to write tombstones?

Also maybe allocate the tombs array if we know we will not be re-writing the block.

Reply:
The dmap length for allocation of tombs I think..

Reply:
Can't really allocate to the length of dmap since it's not related to the amount of tombstones that need to be written. Any tombstones that do need to be written, are not deleted so are also not part of the dmap.

Think it's best to leave as-is for now, given it's the same implementation as what we already have in `PurgeEx` as well. Can look at improving performance if needed later.